logs_analyzer/chapter1/scala/src/main/scala/com/databricks/apps/logs/ApacheAccessLog.scala

```diff
@@ -1,23 +1,35 @@
 package com.databricks.apps.logs
 
-case class ApacheAccessLog(ipAddress: String, clientIdentd: String,
-                           userId: String, dateTime: String, method: String,
-                           endpoint: String, protocol: String,
-                           responseCode: Int, contentSize: Long) {
-
+/**
+  * An entry of Apache access log.
+  */
+case class ApacheAccessLog(ipAddress: String,
+                           clientIdentd: String,
+                           userId: String,
+                           dateTime: String,
+                           method: String,
+                           endpoint: String,
+                           protocol: String,
+                           responseCode: Int,
+                           contentSize: Long) {
 }
 
 object ApacheAccessLog {
   val PATTERN = """^(\S+) (\S+) (\S+) \[([\w:/]+\s[+\-]\d{4})\] "(\S+) (\S+) (\S+)" (\d{3}) (\d+)""".r
 
+  /**
+    * Parse log entry from a string.
+    *
+    * @param log A string, typically a line from a log file
+    * @return An entry of Apache access log
+    * @throws RuntimeException Unable to parse the string
+    */
   def parseLogLine(log: String): ApacheAccessLog = {
```

Comment:
If I understand correctly, `unapply` aims to extract components. If I had `ApacheAccessLog` on input, I could extract `ipAddress`, `clientIdentd` and other parameters via `unapply`:
```
val log: ApacheAccessLog = ...
log match {
  case ApacheAccessLog(ip, client, _, _, _, _, _, _, _)
    => println(s"Client $client from $ip")
  case _ => ...
}
```
However, in this case I do not extract components of `ApacheAccessLog`. Contrarily, I construct an instance of `ApacheAccessLog` from a string. Could you explain, how `unapply` should be used in this case?